### PR TITLE
CIS-1451 Fix channel message visiblity

### PR DESCRIPTION
**Pull request by tbarbugli**
Fixes a regression from the deleted replies fix, the PR adds better tests and cleans up the conditions for showing messages in channels.

### CODE REVIEW

File: Sources/StreamChat/Database/DTOs/MessageDTO_Tests.swift [hunk 1]
```diff
@@ -1233,100 +1233,131 @@ class MessageDTO_Tests: XCTestCase {
...
-        }
-
-        XCTAssertEqual(retrievedMessages.count, 2)
-        XCTAssertEqual(
-            retrievedMessages.map(\.id),
-            [messageId, existingReplyInChannelId]
+        XCTAssertTrue(try checkChannelMessagesPredicate(channelId: channelId, message: message))
+    }
+    
+    func test_channelMessagesPredicate_shouldIncludeDeletedMessages() throws {
```
Comment (polqf) [inline Sources/StreamChat/Database/DTOs/MessageDTO_Tests.swift:1349]: Should we also add a test for thread reply being deleted with showReplyInChannel: true?

**Review by polqf — APPROVED** [review]

**Comment by polqf** [comment]
Also, it is was pointing to develop, but should be pointing to release. Fixed that, but needs a rebase

**Comment by polqf** [comment]
🐛 GIFs coming from giphy don’t appear in main chat independently of what you select

**Review by polqf — CHANGES_REQUESTED** [review]
🐛 GIFs coming from giphy don’t appear in main chat independently of what you select

**Review by nuno-vieira — APPROVED** [review]
QA'd, LGTM ✅

**Comment by polqf** [comment]
Confirmed it is a preexisting issue, we'll move forward

**Review by polqf — APPROVED** [review]

**Pull request merged**